srtcore/socketconfig.h

```diff
@@ -0,0 +1,1140 @@
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*****************************************************************************/
+
+/*****************************************************************************
+written by
+   Haivision Systems Inc.
+*****************************************************************************/
+
+#ifndef INC_SRT_SOCKETCONFIG_H
+#define INC_SRT_SOCKETCONFIG_H
+
+#ifdef SRT_ENABLE_BINDTODEVICE
+#include <linux/if.h>
+#endif
+#include <string>
+#include "haicrypt.h"
+#include "congctl.h"
+#include "packet.h"
+#include "handshake.h"
+#include "logger_defs.h"
+
+// SRT Version constants
+#define SRT_VERSION_UNK     0
+#define SRT_VERSION_MAJ1    0x010000            /* Version 1 major */
+#define SRT_VERSION_MAJ(v) (0xFF0000 & (v))     /* Major number ensuring backward compatibility */
+#define SRT_VERSION_MIN(v) (0x00FF00 & (v))
+#define SRT_VERSION_PCH(v) (0x0000FF & (v))
+
+// NOTE: SRT_VERSION is primarily defined in the build file.
+extern const int32_t SRT_DEF_VERSION;
+
+struct CSrtMuxerConfig
+{
+    static const int
+        DEF_UDP_BUFFER_SIZE = 65536;
+
+    int m_iIpTTL;
+    int m_iIpToS;
+    int m_iIpV6Only;                             // IPV6_V6ONLY option (-1 if not set)
+    bool m_bReuseAddr;                           // reuse an exiting port or not, for UDP multiplexer
+
+#ifdef SRT_ENABLE_BINDTODEVICE
+    std::string m_BindToDevice;
+#endif
+    int m_iUDPSndBufSize;                        // UDP sending buffer size
+    int m_iUDPRcvBufSize;                        // UDP receiving buffer size
+
+    // Some shortcuts
+#define DEFINEP(psym) \
+        char* pv##psym () { return (char*)&m_i##psym; } \
+        const char* pv##psym () const { return (char*)&m_i##psym; }
+
+    DEFINEP(UDPSndBufSize);
+    DEFINEP(UDPRcvBufSize);
+    DEFINEP(IpV6Only);
+    DEFINEP(IpTTL);
+    DEFINEP(IpToS);
+#undef DEFINEP
+
+    bool compatWith(const CSrtMuxerConfig& other)
+    {
+#define CEQUAL(field) (field == other.field)
+        return CEQUAL(m_iIpTTL)
+            && CEQUAL(m_iIpToS)
+            && CEQUAL(m_iIpV6Only)
+            && CEQUAL(m_bReuseAddr)
+#ifdef SRT_ENABLE_BINDTODEVICE
+            && CEQUAL(m_BindToDevice)
+#endif
+            && CEQUAL(m_iUDPSndBufSize)
+            && CEQUAL(m_iUDPRcvBufSize);
+#undef CEQUAL
+    }
```

Comment:
Maybe `isEquelTo(..)` would be better, as the word "compatible" does not suggest equal values.

Reply:
The original intention was to check if the found multiplexer is reusable. I was unaware that it will turn into checking if all fields are equal. OTOH the `isEqualTo` (or even `operator==`) would suggest that it checks if this object is equal (perfectly) to the other multiplexer, while this isn't exactly true because it checks only some of the fields, not all of them.

Reply:
Well, that could have been earlier true - now it compares all fields.